src/librustc_codegen_llvm/consts.rs

```diff
@@ -437,7 +440,18 @@ impl StaticMethods for CodegenCx<'ll, 'tcx> {
                 //
                 // We could remove this hack whenever we decide to drop macOS 10.10 support.
                 if self.tcx.sess.target.target.options.is_like_osx {
-                    let sect_name = if alloc.bytes.iter().all(|b| *b == 0) {
+                    let all_bytes_range = 0..alloc.len();
+                    let is_zeroed = alloc.relocations.len() == 0 && {
```

Comment:
This `alloc.relocations.len() == 0` is new, right? I was told before that this is already inside a section that checks that there are no relocations. So maybe make this an assertion instead?